Library/ViewModels/ConfirmDetailsViewModel.swift

```diff
@@ -0,0 +1,264 @@
+      selectedShippingRule.skipNil(),
+      rewards,
+      selectedQuantities
+    )
+    .map(calculateShippingTotal)
+
+    let baseRewardShippingTotal = Signal.zip(project, baseReward, selectedShippingRule)
+      .map(getBaseRewardShippingTotal)
+
+    let allRewardsShippingTotal = Signal.merge(
+      baseRewardShippingTotal,
+      calculatedShippingTotal
+    )
+
+    // Initial pledge amount is zero if not backed.
+    let initialAdditionalPledgeAmount = Signal.merge(
+      initialData.filter { $0.project.personalization.backing == nil }.mapConst(0.0),
+      backing.map(\.bonusAmount)
+    )
+    .take(first: 1)
+
+    let projectAndReward = Signal.zip(project, baseReward)
+
+    // MARK: Local Pickup + Shipping
+
+    /**
+     Shipping location selector is hidden if the context hides it,
+     if the base reward has no shipping, when add-ons were selected or when base reward has local pickup option.
+     */
+    let nonLocalPickupShippingLocationViewHidden = Signal.combineLatest(baseReward, rewards, context)
+      .map { baseReward, rewards, context in
+        [
+          context.shippingLocationViewHidden,
+          !baseReward.shipping.enabled,
+          rewards.count > 1
+        ].contains(true)
+      }
+
+    self.shippingLocationViewHidden = Signal
+      .combineLatest(nonLocalPickupShippingLocationViewHidden, baseReward)
+      .map { flag, baseReward in
+        isRewardLocalPickup(baseReward) ? true : flag
+      }
+
+    /**
+     Shipping summary view is hidden when updating,
+     if the base reward has no shipping, when NO add-ons were selected or when base reward has local pickup option.
+     */
+    let nonLocalPickupShippingSummaryViewHidden = Signal.combineLatest(baseReward, rewards, context)
+      .map { baseReward, rewards, context in
+        [
+          context.isAny(of: .update, .changePaymentMethod, .fixPaymentMethod),
+          !baseReward.shipping.enabled,
+          rewards.count == 1
+        ].contains(true)
+      }
+
+    self.shippingSummaryViewHidden = Signal.combineLatest(nonLocalPickupShippingSummaryViewHidden, baseReward)
+      .map { flag, baseReward in
+        isRewardLocalPickup(baseReward) ? true : flag
+      }
+
+    let shippingViewsHidden: Signal<Bool, Never> = Signal.combineLatest(
+      self.shippingSummaryViewHidden,
+      self.shippingLocationViewHidden
+    )
+    .map { a, b -> Bool in
+      let r = a && b
+      return r
+    }
+
+    self.localPickupViewHidden = baseReward.map(isRewardLocalPickup).negate()
+
+    // Only shown for regular non-add-ons based rewards
+    self.configureShippingLocationViewWithData = Signal.combineLatest(
+      projectAndReward,
+      shippingViewsHidden.filter(isFalse),
+      selectedLocationId
+    )
+    .map { projectAndReward, _, selectedLocationId in
+      (projectAndReward.0, projectAndReward.1, selectedLocationId)
+    }
+    .map { project, reward, locationId in
+      (project, reward, true, locationId)
```

Comment:
Nit: I've been finding the use of large tuples in checkout to be confusing. Could this be a small wrapper object, instead?